src/gui/entry/EntryURLModel.cpp

```diff
@@ -68,31 +69,43 @@ QVariant EntryURLModel::data(const QModelIndex& index, int role) const
     const auto value = m_entryAttributes->value(key);
     const auto urlValid = Tools::checkUrlValid(value);
 
-    if (role == Qt::BackgroundRole && !urlValid) {
+    // Check for duplicate URLs in the attribute list
+    auto customKeys = m_entryAttributes->customKeys();
+    customKeys.removeOne(key);
+    const auto values = m_entryAttributes->values(customKeys);
+    const auto duplicateUrl = values.contains(value) || value == m_entryUrl;
```

Comment:
I read this code 4 times and couldn't figure out how /why it works. I think the variable names are confusing and it is doing some obtuse list management work to check for duplicates. Can you smooth this out a little?

Reply:
I can smooth it out. Basically it does the following:
- Retrieves all custom keys from the entry
- Removes the current key from the list if it exists (we might be editing an existing URL)
- Get all the values (except the current key/value, because it would always return true)
- Check if the same value is already in the list